core/modules/wmgr/python/procMgr.py

```diff
@@ -0,0 +1,146 @@
+"""
+
+#--------------------------------
+#  Imports of standard modules --
+#--------------------------------
+import logging
+import os
+import subprocess
+
+#-----------------------------
+# Imports for other modules --
+#-----------------------------
+from .config import Config
+from .errors import ExceptionResponse
+from flask import Blueprint, json, request, url_for
+
+#----------------------------------
+# Local non-exported definitions --
+#----------------------------------
+
+_log = logging.getLogger('procMgr')
+
+# static list of service names for now
+_svcNames = ['xrootd', 'mysqld']
+
+def _svcDict(svc, running=None):
+    """ Make service instance dict out of service name """
+    svcDict = dict(name=svc, uri=url_for('.state', service=svc))
+    if running is not None:
+        svcDict['state'] = 'active' if running else 'stopped'
+    return svcDict
+
+def _runCmd(cmd, noexcept=True):
+    """ Run command in a shell """
+
+    try:
+        _log.debug('executing command: %s', cmd)
+        subprocess.check_call(cmd, close_fds=True)
+        return 0
+    except subprocess.CalledProcessError as exc:
+        _log.debug('subprocess exception: %s', exc)
+        if noexcept:
+            return exc.returncode
+        else:
+            # re-throw
+            raise
+
+#------------------------
+# Exported definitions --
+#------------------------
+
+procService = Blueprint('procService', __name__, template_folder='procService')
+
+
+@procService.route('', methods=['GET'])
+def services():
+    """ Return the list of services """
+
+    _log.debug('request: %s', request)
+
+    services = [_svcDict(svc) for svc in _svcNames]
+
+    return json.jsonify(results=services)
+
+
+@procService.route('/<service>', methods=['GET'])
+def state(service):
+    """
+    Return service state.
+    """
+
+    _log.debug('request: %s', request)
+
+    if service not in _svcNames:
+        raise ExceptionResponse(404, "InvalidService", "Invalid service name " + service)
+
+    runDir = Config.instance().runDir
+    initScript = os.path.join(runDir, 'etc/init.d', service)
+
+    cmd = [initScript, 'status']
+    running = _runCmd(cmd) == 0
+
+    return json.jsonify(results=_svcDict(service, running))
+
+
+@procService.route('/<service>', methods=['PUT'])
+def action(service):
+    """
+    Do something with service.
+
+    Following parameters are expected to come in a request (in request body
+    with application/x-www-form-urlencoded content like regular form):
+        action: one of 'stop', 'start', 'restart' (required)
+    """
+
+    _log.debug('request: %s', request)
+
+    if service not in _svcNames:
+        raise ExceptionResponse(404, "InvalidService", "Invalid service name " + service)
+
+    action = request.form.get('action', '').strip()
+    if not action:
+        raise ExceptionResponse(400, "MissingArgument", "Action argument (action) is missing")
+    if action not in ['start', 'stop', 'restart']:
```

Comment:
My mistake, the line below should raise different exception (InvalidArgument instead of MissingArgument). I'll fix it.